traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java

```diff
@@ -52,12 +60,84 @@ public static AuthenticatedUrlRequest catalog(Session session, CatalogQuery cata
         return routeBuilder.build(session.getToken());
     }
 
-    // /product/{id}
-    public static AuthenticatedUrlRequest product(Session session, Product product, Context context) {
-        return new RouteBuilder(context, session.getApiKey())
+    // /booking
+    public static AuthenticatedUrlRequest searchBookingItems(Session session, BookingItemQuery bookingItemQuery, Context context) {
+        RouteBuilder rb = new RouteBuilder(context, session.getApiKey())
                 .method(NetworkTask.Route.Method.GET)
-                .path("/product/" + product.getId())
-                .build(session.getToken());
+                .path("/booking/")
+                .param("text", bookingItemQuery.getQueryText())
+                .param("skip", String.valueOf(bookingItemQuery.getOffset()))
+                .param("take", String.valueOf(bookingItemQuery.getLimit()));
+
+
+        for (ProductItemCategory category : bookingItemQuery.getCategories()) {
+            rb.param("categories", category.toString());
+        }
+
+        if (bookingItemQuery.getPriceRangeFilter() != null) {
+            PriceRangeFilter priceRangeFilter = bookingItemQuery.getPriceRangeFilter();
+            rb.param("min-price", String.valueOf(priceRangeFilter.getRange().getLower()));
+            rb.param("max-price", String.valueOf(priceRangeFilter.getRange().getUpper()));
+            rb.param("currency", Currency.getCode(priceRangeFilter.getCurrency()));
+        }
+
+        if (bookingItemQuery.getBoundingBox() != null) {
+            BoundingBox boundingBox = bookingItemQuery.getBoundingBox();
+            Coordinate topLeftCoordinate = boundingBox.getTopLeftCoordinate();
+            Coordinate bottomRightCoordinate = boundingBox.getBottomRightCoordinate();
+            rb.param("top-left-latitude", String.valueOf(topLeftCoordinate.getLatitude()));
+            rb.param("top-left-longitude", String.valueOf(topLeftCoordinate.getLongitude()));
+            rb.param("bottom-right-latitude", String.valueOf(bottomRightCoordinate.getLatitude()));
+            rb.param("bottom-right-longitude", String.valueOf(bottomRightCoordinate.getLongitude()));
+        }
+
+        return rb.build(session.getToken());
+    }
+
+    // /booking/{id}
+    public static AuthenticatedUrlRequest booking(Session session, Product product, Context context) {
```

Comment:
better naming here....`bookingItem` is good...`booking` is ambiguous a little

Reply:
this is actually part of the previous migration pr. I put the wrong base for this PR. I'll make the fix in the previous version 👍